tests/dbshell/tests.py

```diff
@@ -15,3 +16,25 @@ def test_command_missing(self):
         with self.assertRaisesMessage(CommandError, msg):
             with mock.patch('subprocess.run', side_effect=FileNotFoundError):
                 call_command('dbshell')
+
+
+class BaseDatabaseClientTestCase(SimpleTestCase):
+    def test_runshell_pass_environ(self):
+        args = object()
+        client = BaseDatabaseClient(mock.Mock())
+        with mock.patch('subprocess.run') as mocked:
+            with mock.patch.object(
+                BaseDatabaseClient,
+                'settings_to_cmd_args_env',
+                return_value=(args, None),
+            ):
+                client.runshell(None)
+            mocked.assert_called_once_with(args, env=None, check=True)
+            mocked.reset_mock()
+            with mock.patch.object(
+                BaseDatabaseClient,
+                'settings_to_cmd_args_env',
+                return_value=(args, {}),
+            ):
+                client.runshell(None)
+            mocked.assert_called_once_with(args, env=None, check=True)
```

Comment:
Please use `subTest()`:
```suggestion
        for env in [None, {}]:
            with self.subTest(env=env):
                with mock.patch('subprocess.run') as run:
                    with mock.patch.object(
                        BaseDatabaseClient,
                        'settings_to_cmd_args_env',
                        return_value=(args, env),
                    ):
                        client.runshell(None)
                    run.assert_called_once_with(args, env=None, check=True)
```